includes/class-amp-enhancements.php

```diff
@@ -29,6 +29,38 @@ function ( $gtag_opt ) {
 				return $gtag_opt;
 			}
 		);
+		add_filter( 'should_use_amp_plus', [ __CLASS__, 'amp_plus_mode' ] );
```

Comment:
This logic that determines whether we're in AMP plus or not is sort-of hard to follow between the two plugins (Newspack Plugin and Newspack Ads) and I think it'll get confusing as we add more contexts.

I'd recommend a slightly different approach:

```
public static function should_use_amp_plus( $context = null ) {
    $should = false;

    if ( isset( $_GET['ampplus'] ) && defined( 'NEWSPACK_AMP_PLUS_CONFIG' ) && is_array( NEWSPACK_AMP_PLUS_CONFIG ) && in_array( $context, $config ) ) {
        $should = true;
    }

    return apply_filters( 'should_use_amp_plus', $should, $context );
}
```

I'm also not sure what the behavior of this should be when no context is provided (e.g. `should_use_amp_plus()`). `null` will never be contained in `NEWSPACK_AMP_PLUS_CONFIG`, so that'll always return `false`.

Reply:
Related: https://github.com/Automattic/newspack-ads/pull/81#discussion_r520210980

Reply:
Much nicer, thanks for the idea! https://github.com/Automattic/newspack-plugin/pull/688/commits/9463501b5fd7ab01e8f8b616ff5cc47c54df0411